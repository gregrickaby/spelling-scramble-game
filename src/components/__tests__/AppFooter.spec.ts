import AppFooter from '@/components/AppFooter.vue'
import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'

/**
 * Define a test suite.
 *
 * @see https://vitest.dev/api/#describe
 */
describe('AppFooter', () => {
  /**
   * Mount the component.
   *
   * @see https://vue-test-utils.vuejs.org/api/#mount
   */
  const wrapper = mount(AppFooter)

  /**
   * Assert the component renders.
   *
   * @see https://vitest.dev/api/expect.html
   */
  it('renders component', () => {
    expect(wrapper).toBeTruthy()
  })

  /**
   * Assert the component matches the snapshot.
   *
   * @see https://vitest.dev/api/expect.html#tomatchsnapshot
   */
  it('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
