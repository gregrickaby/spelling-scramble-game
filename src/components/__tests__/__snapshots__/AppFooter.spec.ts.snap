// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppFooter > matches snapshot 1`] = `
"<footer class="flex items-center justify-center gap-4">
  <p> Â©2024 Spelling Scramble </p>
</footer>"
`;

exports[`AppHeader > matches snapshot 1`] = `
"<header class="flex flex-col items-center justify-center gap-4">
  <h1>Spelling Scramble</h1>
  <nav>
    <routerlink to="/">Home</routerlink>
  </nav>
</header>"
`;
